packages/cli/src/util/proposerConfig.ts

```diff
@@ -89,6 +93,8 @@ function parseProposerConfigSection(
     builder: {
       gasLimit: overrideConfig?.builder?.gasLimit ?? (gas_limit !== undefined ? Number(gas_limit) : undefined),
       selection: overrideConfig?.builder?.selection ?? builderSelection,
+      boostFactor:
+        overrideConfig?.builder?.boostFactor ?? (boost_factor !== undefined ? BigInt(boost_factor) : undefined),
```

Comment:
This is the same parsing we had previously for the CLI flag but it produces unclear error if a non-integer value is passed. We should ideally reuse parsing function introduces in https://github.com/ChainSafe/lodestar/pull/6332.

https://github.com/ChainSafe/lodestar/blob/01d47b9a0901c9f03a01ef68c819f4e592cbfd8b/packages/cli/src/cmds/validator/handler.ts#L290

or something along those lines